src/Symfony/Component/PropertyAccess/StringUtil.php

```diff
@@ -197,7 +202,7 @@ public static function singularify($plural)
         }
 
         // Convert teeth to tooth, feet to foot
-        if (false !== ($pos = strpos($plural, 'ee')) && strlen($plural) > 3) {
+        if (false !== ($pos = strpos($plural, 'ee')) && strlen($plural) > 3 ! in_array($plural, self::$allowedEes)) {
```

Comment:
@TomasVotruba for the second comment ? Use an array with lowercase strings in it, and apply strtolower on the plural before the check